backend/business-partner-agent/src/main/java/org/hyperledger/bpa/impl/aries/AriesProofTemplateConditionOperatorFactory.java

```diff
@@ -0,0 +1,194 @@
+            return false;
+        }
+    }
+
+    AttributeAndAttributeGroupOperator schemaIdRestrictionOperator() {
+        return (proofRequestBuilder, name, value) -> proofRequestBuilder.putRestriction(name,
+                restrictions -> restrictions.schemaId(value));
+    }
+
+    AttributeAndAttributeGroupOperator issuedByRestrictionOperator() {
+        return (proofRequestBuilder, name, value) -> proofRequestBuilder.putRestriction(name,
+                restrictions -> restrictions.issuerDid(value));
+    }
+
+    AttributeAndAttributeGroupOperator nonRevokedOperator(Clock clock) {
+        return (proofRequestBuilder, name, value) -> {
+            if (name == null) {
+                proofRequestBuilder.onRequest(
+                        builder -> {
+                            setNonRevoked(clock, value, builder::nonRevoked);
+                            return builder;
+                        });
+            } else {
+                proofRequestBuilder.onAttribute(name, attr -> setNonRevoked(clock, value, attr::nonRevoked)
+                        .orElseGet(() -> {
+                            log.error("Non-Revocation check was not added to " + name);
+                            return attr;
+                        }));
+                proofRequestBuilder.onPredicate(name, pred -> setNonRevoked(clock, value, pred::nonRevoked)
+                        .orElseGet(() -> {
+                            log.error("Non-Revocation check was not added to " + name);
+                            return pred;
+                        }));
+            }
+        };
+    }
+
+    private <T> Optional<T> setNonRevoked(@NonNull Clock clock,
+            @Nullable String value,
+            @NonNull Function<PresentProofRequest.ProofRequest.ProofNonRevoked, T> nonRevokedTarget) {
+        Optional<PresentProofRequest.ProofRequest.ProofNonRevoked> nonRevoked = Optional.empty();
+        try {
+            Long longValue = Optional.ofNullable(value)
+                    .map(Long::decode)
+                    .orElseGet(() -> Math.floorDiv(clock.millis(), 1000));
+            nonRevoked = Optional.of(
+                    PresentProofRequest.ProofRequest.ProofNonRevoked.builder()
+                            .from(longValue)
+                            .to(longValue)
+                            .build());
+        } catch (NumberFormatException e) {
+            log.error("non revocation operator need a number value.", e);
+        }
+        return nonRevoked.map(nonRevokedTarget);
+    }
+
+    AttributeCreatingOperator fetchValueOperator() {
+        return (proofRequestBuilder, name, value) -> proofRequestBuilder.addAttribute(name);
+    }
+
+    AttributeCreatingOperator equalsOperator() {
+        return (proofRequestBuilder, name, value) -> {
+            if (name == null || value == null) {
+                throw new RuntimeException("equals conditions require an attribute name and a value.");
```

Comment:
This will result in an internal server error. Is this expected behavior?

Reply:
Now that you point that out, I don't not think, it's the preferred behavior. I wanted reject misconfigured templates. But at the time this line is executed, the template was already successfully stored and is being transformed into a proof request.

I have find another approach for this.